cb-paradise-project-frontend/eureka-v1.1.0
# Improve Modal based on #25 PR

Achieved:
1. add an experimental "ModalTest" component, its style and structure is based on existing "Modal" and "Overlay" designed by @ttcode10 , so no longer depends on react-modal package;
2. use createPortal to append modals on top-level instead of using "ModalContext";
3. use routes instead of states to toggle modals;
4. add optional AlertModal which can be triggered when closing the main modal;
5. extract Navigation from Landing for testing modals on different pages;
6. refactor Routes in the App component to solve path conflicts;

How To Use the "ModalTest" Component:
1.	create a page with ModalPage component or your customized layout;
2.	import ModalTest component to wrap the page and export them;
3.	import the page, and set a route for the modal in App component, the path should match the end of the current URL;
4.	use a link to point to the modal's route, the link should base on the current path;
5.	(optional) set "comfirmeBeforeClose" attribute to activate the alert modal which asks for your confirmation before closing the main modal;
